springwolf-plugins/springwolf-cloud-stream-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/cloudstream/FunctionalChannelBeanData.java

```diff
@@ -79,8 +79,16 @@ private static Class<?> toClassObject(Type type) {
             if ("org.apache.kafka.streams.kstream.KStream".equals(rawType.getTypeName())) {
                 return (Class<?>) ((ParameterizedType) type).getActualTypeArguments()[1];
             }
+            // Resolve generic type for batch listeners
+            if (rawType == List.class) {
```

Comment:
This will not solve your issue - but it is close. Instead of checking for raw type  equal to `List`, you should check equality to `Message`. 

You could also improve the readability and maintainability of this code by extracting the handling of this case to another method.